python/taichi/lang/matrix_ops.py

```diff
@@ -55,18 +55,25 @@ def _reduce(mat, fun: template()):
         foreach(
             Or(assert_vector(),
                assert_list,
-               msg="Cols/rows must be a list of lists, or a list of vectors")),
-        same_shapes))
+               msg="Cols/rows must be a list of lists, or a list of vectors")))
+)
 def rows(rows):  # pylint: disable=W0621
     if isinstance(rows[0], (Matrix, Expr)):
         shape = rows[0].get_shape()
         assert len(shape) == 1, "Rows must be a list of vectors"
 
+        for i, row in enumerate(rows):
+            ast_builder = get_runtime().prog.current_ast_builder()
+            if current_cfg().real_matrix and isinstance(
+                    row, Expr) and row.is_tensor():
+                rows[i] = [Expr(x) for x in ast_builder.expand_expr([row.ptr])]
```

Comment:
Is this part necessary? IIRC `for row in rows` should be able to work after #6427.

Reply:
Thanks!